src/client.c

```diff
@@ -154,7 +154,7 @@ int hive_client_logout(HiveClient *client)
     return 0;
 }
 
-int hive_client_get_info(HiveClient *client, char **result)
+int hive_client_get_info(HiveClient *client, HiveClientInfo **result)
```

Comment:
HiveClientInfo should be the structure with buffer already. It should be allocated inside functions. for instance, the structure should be like that:
struct HiveClientInfo {
   char clientid[MAX_CLIENT_ID_LENGTH + 1];
   char displayName[MAX_CLIENT_DISPLAY_NAME_LENTH + 1];
   ...
}